### ISSUE TITLE
mod import error with k8s 1.30 and kubevirt 1.3.0

### ISSUE THREAD

**Issue by hemakshis** [kubevirt/kubevirt#12722]
**What happened**:
We are working on upgrading kubernetes in our go projects to v1.30 and as per the [kubevirt support matrix](https://github.com/kubevirt/sig-release/blob/main/releases/k8s-support-matrix.md) we should update kubevirt to v1.3.0 but seeing the following error after updating kubernetes to v1.30 and kubevirt to v1.3.0 I am facing the following issue -

```shell
❯ go mod tidy
go: k8s.io/kube-openapi@v0.30.0: reading k8s.io/kube-openapi/go.mod at revision v0.30.0: unknown revision v0.30.0
```

If I go by k8s v1.29 and kubevirt v1.2.0 then facing the following issue -
```shell
❯ go mod tidy    
go: finding module for package k8s.io/client-go/kubernetes/typed/resource/v1alpha1
go: finding module for package k8s.io/client-go/kubernetes/typed/flowcontrol/v1alpha1
go: github.com/portworx/sched-ops/k8s/kubevirt imports
        kubevirt.io/client-go/kubecli imports
        k8s.io/client-go/kubernetes/typed/flowcontrol/v1alpha1: module k8s.io/client-go@latest found (v0.0.0-00010101000000-000000000000, replaced by k8s.io/client-go@v0.29.8), but does not contain package k8s.io/client-go/kubernetes/typed/flowcontrol/v1alpha1
go: github.com/portworx/sched-ops/k8s/kubevirt imports
        kubevirt.io/client-go/kubecli imports
        k8s.io/client-go/kubernetes/typed/resource/v1alpha1: module k8s.io/client-go@latest found (v0.0.0-00010101000000-000000000000, replaced by k8s.io/client-go@v0.29.8), but does not contain package k8s.io/client-go/kubernetes/typed/resource/v1alpha1
```

**What you expected to happen**:
`go mod tidy` should pass successfully and expect the kubevirt support matrix to have this fixed

**How to reproduce it (as minimally and precisely as possible)**:

**Additional context**:
This bug has already been reported [here](https://github.com/kubevirt/client-go/issues/38) and a PR has also been raised for the same, but looks like the code is maintained in this repository.

**Environment**:
- KubeVirt version (use `virtctl version`): v1.3.0
- Kubernetes version (use `kubectl version`): v1.30.4
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

**Comment by oscarlear** [comment]
modified go.mod like this
```
require (
k8s.io/kube-openapi v0.30.0
)

replace (
 k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20240430033511-f0e62f92d13f
)
```
find `k8s.io/kube-openapi` version in `kubevirt.io/client-go`  go.mod files

**Comment by Forrest-Tao** [comment]
> modified go.mod like this
> 
> ```
> require (
> k8s.io/kube-openapi v0.30.0
> )
> 
> replace (
>  k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20240430033511-f0e62f92d13f
> )
> ```
> 
> find `k8s.io/kube-openapi` version in `kubevirt.io/client-go` go.mod files

thanks,it works

**Comment by JayBusch** [comment]
same..... ish.... library seems unusable in it's current state.

go: k8s.io/kube-openapi@v0.31.0: reading k8s.io/kube-openapi/go.mod at revision v0.31.0: unknown revision v0.31.0

**Comment by tiansuo114** [comment]
> same..... ish.... library seems unusable in it's current state.
> 
> go: k8s.io/kube-openapi@v0.31.0: reading k8s.io/kube-openapi/go.mod at revision v0.31.0: unknown revision v0.31.0

Yes, this issue still persists, and I am unsure how to resolve it. I have tried reaching out through Google groups and community forums for assistance.

Perhaps this issue needs more attention from the community? @vladikr

**Comment by tiansuo114** [comment]
> same..... ish.... library seems unusable in it's current state.
> 
> go: k8s.io/kube-openapi@v0.31.0: reading k8s.io/kube-openapi/go.mod at revision v0.31.0: unknown revision v0.31.0

In the go.mod file, you found the following code:

```
require (
    k8s.io/kube-openapi v0.31.0
)

replace (
    k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20240430033511-f0e62f92d13f
)
```
I added the same code to your own project, which seems to have resolved the issue. However, you are still unclear about the purpose of doing so.

**Comment by cade-thomas** [comment]
Does anyone have a currently working work around for this?
